packages/dashboard-core-plugins/src/panels/MarkdownNotebook.tsx

```diff
@@ -1,15 +1,55 @@
-import React, { PureComponent } from 'react';
+import React, {
+  PureComponent,
+  ReactElement,
+  RefObject,
+  MouseEvent,
+  MouseEventHandler,
+} from 'react';
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import Markdown from 'react-markdown';
 import remarkGfm from 'remark-gfm';
 import { Button } from '@deephaven/components';
 import { Code } from '@deephaven/console';
 import { vsPlay } from '@deephaven/icons';
 import './MarkdownNotebook.scss';
+import {
+  ReactMarkdownProps,
+  TransformImage,
+  TransformLink,
+} from 'react-markdown/src/ast-to-react';
+import { assertNotNull } from '@deephaven/utils';
 
-export class MarkdownNotebook extends PureComponent {
-  constructor(props) {
+interface MarkdownNotebookProps {
+  onRunCode: (command?: string) => void;
+  content: string;
+  onLinkClick: MouseEventHandler<HTMLAnchorElement>;
+  transformImageUri?: TransformImage;
+  transformLinkUri?: false | TransformLink | null;
+}
+
+interface MarkdownNotebookState {
+  hasCode: boolean;
+
+  // Keep track if any code has been executed yet. If not, make the run button flash
+  hasRunCode: boolean;
+
+  // Line of the next block to execute. Null to start at the first block
+  nextStartLine: number | null;
+}
+
+export class MarkdownNotebook extends PureComponent<
+  MarkdownNotebookProps,
+  MarkdownNotebookState
+> {
+  static defaultProps = {
+    content: '',
+    onLinkClick: undefined,
+    onRunCode: (): void => undefined,
+    transformImageUri: undefined,
+    transformLinkUri: undefined,
```

Comment:
No need to set `defaultProps` to `undefined` on optional params:
```suggestion
    onRunCode: (): void => undefined,
```